### ISSUE TITLE
Is there a better way to organize the wrapped sharing API 

### ISSUE THREAD

**Issue by bilowik** [bilowik/sss-rs#5]
I've always hated how Secret works, but at the time and even now I still am not sure of a better way to handle it. I would think that sharing a large file would be a common use of this library which is why I want to include an easy way to share a large file. 

At the very least, I might have the reconstruct functions be defined outside of the secret and just take a secret for consistency with the the sharing functions. This is of course a breaking change but it's a needed one. 

Anyone else who swings by and has an opinion absolutely respond, there might be something stupidly obvious that I'm just not seeing lol. 

It would also be nice to prevent double-reconstructing into the same secret at compile time but I don't believe there is a way to do this.

**Comment by bilowik** [comment]
Have a branch with a change to make the share/reconstruct more consistent with one another, breaking change obviously, but like I said above, a needed one. Holding off on it to see if I can't come up with something better.

**Comment by bilowik** [comment]
The more fixes I tried the more i realized why I did it this way in the first place lol. Because of needing to calculate the hash we would need T: Write/Read + Seek for sharing and reconstructing, which would mean we'd need to wrap vec secrets in a cursor every time. Maybe this is better? I don't know :/

**Comment by bilowik** [comment]
Reopening bc I still fucking hate how the wrapped sharing api works. Randomly popped back into my head and I immediately realized a better API.

**Comment by bilowik** [comment]
Why not a generic parameter T: Read for sharing and a T: Write for reconstructing? 

```rust
fn reconstruct<T: Write>(out: T, srcs: Vec<Vec<u8>>,  verify: bool) -> Result;
```
instead of 
```rust
fn reconstruct(secret: &mut Secret, srcs: Vec<Vec<u8>>,  verify: bool) -> Result;
```

And for backwards compat, implement Read and Write onto Secret, which should be straightforward since Secret is made up of either a file name or a vec which can be wrapped in a Cursor. 

And maybe an convenience function called reconstruct_to_vec which just wraps around the new reconstruct by creating a vec and returning it. Something like:
```rust
fn reconstruct_to_vec(srcs: Vec<Vec<u8>>,  verify: bool) -> Result<Vec<u8>>;
```

Fully backwards compatible, easier to use, sounds like a win.

**Comment by bilowik** [comment]
Ah just read some of the original comments, I realize the issue I had the first time I tried this. The only place I can really see Read + Write + Seek being an issue is network streams. But I can't think of a great case for reconstructing from network streams, not to mention it isn't currently any easier/harder  so I don't see a reason not to go with this.

**Comment by bilowik** [comment]
Implemented with 0.10.0
